x/emissions/keeper/msg_server.go

```diff
@@ -741,6 +741,96 @@ func (ms msgServer) RequestInference(ctx context.Context, msg *state.MsgRequestI
 	return &state.MsgRequestInferenceResponse{}, nil
 }
 
+func (ms msgServer) RequestSingleInference(ctx context.Context, msg *state.MsgRequestSingleInference) (*state.MsgRequestInferenceResponse, error) {
```

Comment:
you can make this even simpler, you can can construct a MsgRequestInference inside this function and call RequestInference directly from RequestSingleInference